driver/src/contracts/stablex_contract.rs

```diff
@@ -59,12 +59,74 @@ impl StableXContract for StableXContractImpl {
 
     fn submit_solution(
         &self,
-        _batch_index: U256,
-        _orders: Vec<Order>,
-        _solution: Solution,
+        batch_index: U256,
+        orders: Vec<Order>,
+        solution: Solution,
     ) -> Result<()> {
-        unimplemented!();
+        let account = self
+            .base
+            .account_with_sufficient_balance()
+            .ok_or("Not enough balance to send Txs")?;
+
+        let (owners, order_ids, volumes, prices, token_ids_for_price) =
+            parse_auction_results(orders, solution);
+
+        self.base
+            .contract
+            .call(
+                "submitSolution",
+                (
+                    batch_index,
+                    owners,
+                    order_ids,
+                    volumes,
+                    prices,
+                    token_ids_for_price,
+                ),
+                account,
+                Options::default(),
+            )
+            .wait()
+            .map_err(DriverError::from)
+            .map(|_| ())
+    }
+}
+
+type ContractRecognizedAuctionResults = (Vec<H160>, Vec<U128>, Vec<U128>, Vec<U128>, Vec<U128>);
+
+fn parse_auction_results(
+    orders: Vec<Order>,
+    solution: Solution,
+) -> ContractRecognizedAuctionResults {
+    // Representing the solution's price vector more compactly as:
+    // sorted_touched_token_ids, non_zero_prices which are logically bound by index.
+    // Example solution.prices = [3, 0, 1] will be transformed into [0, 2], [3, 1]
+    let mut ordered_token_ids: Vec<U128> = vec![];
+    let mut prices: Vec<U128> = vec![];
+    for (token_id, price) in solution.prices.into_iter().enumerate() {
```

Comment:
Yes, at some point (if we every reach 2^16 ~ 65K tokens) it will be imperative that we do not iterate (even once) over this list. Back when we were using 2^8 this would be no problem at all! Ha ha.

Reply:
I will create a "task" for this in our backlog and perhaps include a `todo` here in the code.